domain/src/test/java/org/oppia/android/domain/classify/rules/numberwithunits/NumberWithUnitsIsEqualToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,231 @@
+    InteractionObjectTestBuilder.createFraction(
+      isNegative = false, numerator = 2, denominator = 5
+    ).fraction
+  private val TEST_NUMBER_FRACTION_VALUE_1_OVER_4 =
+    InteractionObjectTestBuilder.createFraction(
+      isNegative = false, numerator = 1, denominator = 4
+    ).fraction
+  private val TEST_REAL_VALUE_DIFFERENT_TYPE =
+    InteractionObjectTestBuilder.createReal(value = 6.9)
+  private val TEST_REAL_VALUE =
+    InteractionObjectTestBuilder.createReal(value = 3.8).real
+  private val TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_5 =
+    InteractionObjectTestBuilder.createNumberUnit(unit = "p", exponent = 5)
+  private val TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_3 =
+    InteractionObjectTestBuilder.createNumberUnit(unit = "e", exponent = 3)
+  private val TEST_NUMBER_WITH_UNITS_ANSWER =
+    InteractionObjectTestBuilder.createNumberWithUnitsForFraction(
+      TEST_NUMBER_FRACTION_VALUE_2_OVER_5,
+      listOf(
+        TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_3,
+        TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_5
+      )
+    )
+  private val TEST_NUMBER_WITH_UNITS_INPUT =
+    InteractionObjectTestBuilder.createNumberWithUnitsForFraction(
+      TEST_NUMBER_FRACTION_VALUE_2_OVER_5,
+      listOf(
+        TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_3,
+        TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_5
+      )
+    )
+  private val TEST_DIFFERENT_NUMBERS_WITH_UNITS =
+    InteractionObjectTestBuilder.createNumberWithUnitsForFraction(
+      TEST_NUMBER_FRACTION_VALUE_1_OVER_4,
+      listOf(
+        TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_5,
+        TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_3
+      )
+    )
+  private val TEST_REAL_INPUT_NUMBER_WITH_UNITS =
+    InteractionObjectTestBuilder.createNumberWithUnitsForReal(
+      TEST_REAL_VALUE,
+      listOf(
+        TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_3,
+        TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_5
+      )
+    )
+  private val TEST_REAL_ANSWER_NUMBER_WITH_UNITS =
+    InteractionObjectTestBuilder.createNumberWithUnitsForReal(
+      TEST_REAL_VALUE,
+      listOf(
+        TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_5,
+        TEST_NUMBER_UNIT_WITH_STRING_TO_POWER_3
+      )
+    )
+
+  @Inject
+  internal lateinit var numberWithUnitsIsEqualToRuleClassifierProvider:
+    NumberWithUnitsIsEqualToRuleClassifierProvider
+
+  private val unitsIsEqualsRuleClassifier by lazy {
+    numberWithUnitsIsEqualToRuleClassifierProvider.createRuleClassifier()
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testEquals_fractionInputNumberWithUnits_withFractionAnswerWithUnits_bothValuesMatches() {
+    val inputs = mapOf("f" to TEST_NUMBER_WITH_UNITS_INPUT)
+
+    val matches =
+      unitsIsEqualsRuleClassifier.matches(
+        answer = TEST_NUMBER_WITH_UNITS_ANSWER,
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquals_differentFractionWithUnits_withRealAnswerNumberWithUnits_bothDoNotMatch() {
```

Comment:
`testDifferentFractionWithUnits_withRealAnswerNumberWithUnits_bothDoNotMatch` this sounds short and similar too.